### ISSUE TITLE
Early stopping does not work

### ISSUE THREAD

**Issue by plumdeq** [ncullen93/torchsample#60]
Hello,

It appears as though the `EarlyStopping`callbacks do not work as they should. Having defined the `ModuleTrainer`as below,

```python
learning_rate = 1e-5
optimizer = torch.optim.Adam(model.parameters(), lr=learning_rate)
loss = torch.nn.BCEWithLogitsLoss(size_average=False)
regularizers = [L2Regularizer(scale=1e-5)]
callbacks = [EarlyStopping(monitor='val_loss', patience=5, min_delta=1)]
optimizer.zero_grad()

trainer = ModuleTrainer(model)
trainer.compile(loss=loss, 
                callbacks=callbacks,
                regularizers=regularizers,
                optimizer=optimizer)
```

the training process does not stop early, even though the validation loss stays the same for far too many epochs (as shown below):

```bash
...
Epoch 4983/5000: 19 batches [00:00, 117.97 batches/s, loss=20.4344, reg_loss=0.0011, val_loss=16.2812]
Epoch 4984/5000: 19 batches [00:00, 116.54 batches/s, loss=20.4344, reg_loss=0.0011, val_loss=16.2812]
Epoch 4985/5000: 19 batches [00:00, 110.11 batches/s, loss=20.4344, reg_loss=0.0011, val_loss=16.2812]
Epoch 4986/5000: 19 batches [00:00, 110.94 batches/s, loss=20.4344, reg_loss=0.0011, val_loss=16.2812]
Epoch 4987/5000: 19 batches [00:00, 105.54 batches/s, loss=20.4344, reg_loss=0.0011, val_loss=16.2812]
Epoch 4988/5000: 19 batches [00:00, 118.59 batches/s, loss=20.4344, reg_loss=0.0011, val_loss=16.2812]
Epoch 4989/5000: 19 batches [00:00, 117.49 batches/s, loss=20.4344, reg_loss=0.0011, val_loss=16.2812]
Epoch 4990/5000: 19 batches [00:00, 108.96 batches/s, loss=20.4344, reg_loss=0.0011, val_loss=16.2812]
Epoch 4991/5000: 19 batches [00:00, 111.27 batches/s, loss=20.4344, reg_loss=0.0011, val_loss=16.2812]
Epoch 4992/5000: 19 batches [00:00, 117.38 batches/s, loss=20.4344, reg_loss=0.0011, val_loss=16.2812]
Epoch 4993/5000: 19 batches [00:00, 118.45 batches/s, loss=20.4344, reg_loss=0.0011, val_loss=16.2812]
Epoch 4994/5000: 19 batches [00:00, 115.60 batches/s, loss=20.4344, reg_loss=0.0011, val_loss=16.2812]
Epoch 4995/5000: 19 batches [00:00, 110.75 batches/s, loss=20.4344, reg_loss=0.0011, val_loss=16.2812]
Epoch 4996/5000: 19 batches [00:00, 111.68 batches/s, loss=20.4344, reg_loss=0.0011, val_loss=16.2812]
Epoch 4997/5000: 19 batches [00:00, 115.34 batches/s, loss=20.4344, reg_loss=0.0011, val_loss=16.2812]
Epoch 4998/5000: 19 batches [00:00, 110.62 batches/s, loss=20.4344, reg_loss=0.0011, val_loss=16.2812]
Epoch 4999/5000: 19 batches [00:00, 109.61 batches/s, loss=20.4344, reg_loss=0.0011, val_loss=16.2812]
Epoch 5000/5000: 19 batches [00:00, 110.49 batches/s, loss=20.4344, reg_loss=0.0011, val_loss=16.2812]
```

Any ideas what am I doing wrong, or there is a bug?

P.s. same story if I add callbacks to the `model` object with `model.set_callback`.
